src/Agent.Sdk/Util/ParentProcessUtil/InteropParentProcessFinder.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Agent.Sdk.Util.ParentProcessUtil
+{
+    internal class Interop
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct PROCESS_BASIC_INFORMATION
+        {
+            internal IntPtr ExitStatus;
+            internal IntPtr PebBaseAddress;
+            internal IntPtr AffinityMask;
+            internal IntPtr BasePriority;
+            internal IntPtr UniqueProcessId;
+            internal IntPtr InheritedFromUniqueProcessId;
+        }
+
+        [DllImport("ntdll.dll", SetLastError = true)]
+        internal static extern int NtQueryInformationProcess(IntPtr processHandle, int processInformationClass, out PROCESS_BASIC_INFORMATION processInformation, int processInformationLength, out int returnLength);
+    }
+
+    internal static class InteropParentProcessFinder
```

Comment:
The comments here are moot if the feedback in Agent.cs can be addressed to use Env variable e.g PSModulePath to check for the issue rather than iterating the parent processes

Reply:
Done, but I did not remove iteration of parent processes completely. Left the WMI one to preserve compatibility for CleanupPsModule knob